build.sbt

```diff
@@ -276,6 +271,35 @@ lazy val V = new {
     ).toList
 }
 
+// -Xlint is unusable because of
+// https://github.com/scala/bug/issues/10448
+def lintingOptions(scalaVersion: String) = {
+  val unused213 = "-Wunused"
+  val unused3 = "-Wunused:all"
+  val common = List(
+    // desugaring of for yield caused pattern var to complain
+    // https://github.com/scala/bug/issues/10287
+    "-Wconf:msg=parameter value .+ in anonymous function:silent",
+    // silence unused parameters in mtags
```

Comment:
Maybe a silly question, but are they false positives? Or what not remove the params if they are unused instead of silencing them?

Reply:
Yes, they're false positives. It's because:
```scala
  for {
    value <- opt
    if value
    result <- computeSth()
  } yield result
```
is being desugared more or less to the:
```scala
  opt
    .withFilter(value => value)
    .flatMap(value =>
      computeSth()
        .map(result => result)
    )
```
and it is true that `value` in flatmap is unused.

I could rewrite them but it's an inadequate amount of work compared to the value it brings.

Reply:
Gotcha, thanks for the explanation.